app/src/main/java/com/waseefakhtar/doseapp/util/TimeUtils.kt

```diff
@@ -0,0 +1,18 @@
+package com.waseefakhtar.doseapp.util
+
+import com.waseefakhtar.doseapp.domain.model.Medication
+import com.waseefakhtar.doseapp.extension.toFormattedString
+import java.util.concurrent.TimeUnit
+import kotlin.math.abs
+
+
+fun getTimeRemaining(medication: Medication): Long {
+    return if (medication.date.toFormattedString() == medication.endDate.toFormattedString()) {
+        0L
+    } else {
+        val dateBeforeInMs: Long = medication.date.time
+        val dateAfterInMs: Long = medication.endDate.time
+        val timeDiff = abs(dateAfterInMs - dateBeforeInMs)
+        (TimeUnit.DAYS.convert(timeDiff, TimeUnit.MILLISECONDS) + 1L)
```

Comment:
Cool mechanism!

Perhaps show the time remaining by hours and minutes if the dose is to be taken on the current date. For future days, we can show days.

Example:

Next dose in %1$s days => If the dose time is not today.
Next dose in %1$s hours => If the dose time is not within the next one hour.
Next dose in %1$s minutes => If the dose time is within an hour.

Reply:
@waseefakhtar 
How to handle the time when the medicine is not taken on the current date.
Because with the current implementation the time shows with the negative when the medicine is not taken on the current day.

Reply:
Good point. My bad that I did not explain it properly. The plan was to store the date in such a way that:

1. If the TimeOfDay is set for Morning, the time is set for 8 AM.
2. If the TimeOfDay is set for Afternoon, the time is set for 1 PM.
3. If the TimeOfDay is set for Evening, the time is set for 6 PM.
4. If the TimeOfDay is set for Night, the time is set for 10 PM.

This way when retrieving the date/time, it is easy to calculate the different from currentTime.

I've now updated the PR and your mechanism with the changes. :)